code/modules/projectiles/guns/energy/cell_loaded_vr/multi_cannon_cells.dm

```diff
@@ -0,0 +1,102 @@
+/obj/item/ammo_casing/macrobattery
+	caliber = "macrobat"
+	name = "macrobattery"
+	icon = 'icons/obj/ammo_vr.dmi'
+	icon_state = "macrobat_wtf"
+	desc = "A large nanite fabricator for a medigun. Powered by a mix of precursor and modern tech, this fancy device recharges without an outside power source."
+
+	projectile_type = /obj/item/projectile/beam/chain_lightning //why the hell not
+
+	var/bat_colour = "#ff33cc"
+	var/charge
+	var/max_charge = 10
+	var/ticks = 1
+	var/ticks_to_charge = 5 //10 secs per shot charged
+
+/obj/item/ammo_casing/macrobattery/Initialize(mapload, ...)
+	. = ..()
+	START_PROCESSING(SSobj, src)
+	charge = max_charge
+
+/obj/item/ammo_casing/macrobattery/Destroy()
+	STOP_PROCESSING(SSobj, src)
+	. = ..()
+	
+/obj/item/ammo_casing/macrobattery/process()
+	ticks++
+	if(ticks%ticks_to_charge == 0)
+		recharge()
+		if(charge >= max_charge)
+			return PROCESS_KILL
+
+/obj/item/ammo_casing/macrobattery/expend()
+	if(charge)
+		charge --
+		ticks = 1 //so we have to start over on the charge time.
+		START_PROCESSING(SSobj, src)
+		update_icon()
+		. = BB
+		//alright, the below seems jank. it IS jank, but for whatever reason I can't reuse BB. big bad
+		BB = null
+		BB = new projectile_type
+		return 
+	else
+		BB = null
+		return null
+
+/obj/item/ammo_casing/macrobattery/proc/recharge()
+	if(charge < max_charge)
+		charge ++
+		if(!BB)
+			BB = new projectile_type
+	if(charge >= max_charge)
+		STOP_PROCESSING(SSobj, src)
+	if(istype(loc,/obj/item/weapon/gun/projectile/multi_cannon))
+		var/obj/item/weapon/gun/projectile/multi_cannon/gun = loc
+		gun.update_icon()	
```

Comment:
```suggestion
	if(loc)
		loc.update_icon()	
```
Here's how normal cells do it. You don't actually need to cast it as anything, because we defined `update_icon()` on atom itself, and that's what `loc` is in byond, so no need to typecheck or recast.

Reply:
The reason I'm typechecking is so it doesn't constantly update whatever it's loc is when charging. I'm sure an extra update_icon call isn't the end of the world, but better to avoid it when possible imo

Reply:
Every cell already does it, sadly. I doubt it'd make that much of a difference.